### ISSUE TITLE
Support GFF3 format

### ISSUE THREAD

**Issue by heuermh** [bigdatagenomics/adam#1007]

**Comment by heuermh** [comment]
For ```bdg-formats``` changes I was thinking
```
record ExternalIdentifier // could replace Dbxref?
{
  string db; // or database, or authority, or namespace
  string id; // or identifier, or accession
  string version;
}
record OntologyTerm
{
  string id; // or identifier
  union { null, string } term = null; // or concept
  union { null, string } sourceName = null; // or authority
  union { null, string } sourceVersion = null;
}
record Feature
{
- array<string> parentIds = [];
+ union { null, string } parentId = null; // not sure about this

+ array<string> childIds = [];

- union { null, string } featureType = null;
+ union { null, OntologyTerm } featureType = null;
}
```
And then some way of adding OntologyTerm and ExternalIdentifier as value types in attributes, either as separate maps
```
record Feature
{
  map<string> attributes = {};

+ map<OntologyTerm> ontologyTerms = {};  // Map<String, OntologyTerm> in avro, right?

- array<Dbxref> dbxrefs = [];
+ map<ExternalIdentifiers> externalIdentifiers = {};
}
```
or something like the Attributes record in GA4GH
```
record Attributes
{
  map<array<union {string, ExternalIdentifier, OntologyTerm}>> values = {};
}
record Feature
{
- map<string> attributes = {};
+ union { null, Attributes } attributes = null;
}
```

**Comment by heuermh** [comment]
FYI the spec was recently moved here
https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md

**Comment by heuermh** [comment]
@tdanford could we have another round of discussion on this one?

**Comment by heuermh** [comment]
Fixed by #998
